Cargo.lock

```diff
@@ -1842,6 +1924,28 @@ dependencies = [
  "getrandom",
 ]
 
+[[package]]
+name = "rayon"
+version = "1.7.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "1d2df5196e37bcc87abebc0053e20787d73847bb33134a69841207dd0a47f03b"
+dependencies = [
+ "either",
+ "rayon-core",
+]
+
+[[package]]
+name = "rayon-core"
+version = "1.11.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "4b8f95bd6966f5c87776639160a66bd8ab9895d9d4ab01ddba9fc60661aebe8d"
+dependencies = [
+ "crossbeam-channel",
+ "crossbeam-deque",
+ "crossbeam-utils",
+ "num_cpus",
+]
```

Comment:
Hmm do we even need rayon? I imagine performance would be good enough even without parallel sort

Reply:
I use rayon in lot int the parallel worker. It's not just the parallel sort but more importantly `par_extend' and a parallel `for_each` and in general parallel iterators (basically running the matcher in parallel). I went down the rabbit hole of replacing rayon a while ago. 

For high-performance parallel code with small units of work like this, I would say rayon is unbeatable. The work-stealing thread pool from rayon core is amazingly performant and well thought out. The parallel iterators are built around dynamically taking advantage of work splitting and are really good at assigning optimal units of work to each thread. Nucleo benefits from that a lot because the matcher has a lot of fast paths built in (like fast rejection) so it's very hard to predict how long each match will take (so if you split the list of matches into N chunks and give N thread one chunk its quite likely that one thread wil&take significantly longer). A good adaptive work-splitting approach like in rayon is just the right thing and benefits nulco a lot.

I looked into just using `rayon-core` and reimplementing just the parts of rayon that I would need but that involves copying a *lot* of (unsafe) code and doesn't feel like a good tradeoff to me.  If you feel uncomfortable pulling in all of rayon I could do that if you see it as a requirement. I would still need 'rayon-core` tough (the work-stealing worker is really good at what it does).